### PR TITLE
PERF-#4860: `deploy_axis_func` should be serialized only once

### LINKED ISSUE #4860: PERF: `PandasDataframeAxisPartition.deploy_axis_func` should be serialized only once
Place: https://github.com/modin-project/modin/blob/cfcdf081e197d35e1fd1526f5585e361793dc2fa/modin/core/execution/ray/implementations/pandas_on_ray/partitioning/virtual_partition.py#L191

Currently, this function is serialized on every call.

### PR DESCRIPTION
**Pull request by anmyachev**
Signed-off-by: Myachev <anatoly.myachev@intel.com>

## What do these changes do?

- [x] commit message follows format outlined [here](https://modin.readthedocs.io/en/latest/development/contributing.html#commit-message-formatting)
- [x] passes `flake8 modin/ asv_bench/benchmarks scripts/doc_checker.py`
- [x] passes `black --check modin/ asv_bench/benchmarks scripts/doc_checker.py`
- [x] signed commit with `git commit -s` 
- [x] Resolves #4860 
- [x] tests added and passing
- [x] module layout described at `docs/development/architecture.rst` is up-to-date 
- [x] added (Issue Number: PR title (PR Number)) and github username to release notes for next major release

### CODE REVIEW

**Review by RehanSD — COMMENTED** [review]
LGTM - do we have any perf benches for this? One concern is that if we have multiple nodes there could be communication overhead since the function will be serialized and placed in one nodes object store, and the other nodes will have to request the function from that nodes object store every time.

File: modin/core/execution/ray/implementations/pandas_on_ray/partitioning/virtual_partition.py [hunk 1]
```diff
@@ -24,6 +24,9 @@
 from .partition import PandasOnRayDataframePartition
 
 
+_DEPLOY_AXIS_FUNC = None
```
Comment (YarShev) [inline modin/core/execution/ray/implementations/pandas_on_ray/partitioning/virtual_partition.py:27]:
```suggestion
_DEPLOY_AXIS_FUNC = ray.put(PandasDataframeAxisPartition.deploy_axis_func)
```
maybe just so?

**Comment by anmyachev** [comment]
> LGTM - do we have any perf benches for this? One concern is that if we have multiple nodes there could be communication overhead since the function will be serialized and placed in one nodes object store, and the other nodes will have to request the function from that nodes object store every time.

No, I see perf gain in my example. The machine that was used for the measurements has 112 cores. With 1000 calls to deploy_axis_func , I see a decrease in time of about 1 second. I don't test the function on several nodes, but doesn't Ray broadcast the function on all node (or at least cache it when first time called)?

**Review by YarShev — APPROVED** [review]

**Comment by vnlitvinov** [comment]
> the other nodes will have to request the function from that nodes object store every time.

Other nodes should request it once and keep in their own object stores as long as there's free space.

That said, I think we should try to use this pattern more (and try to go away from functions defined inside other functions - this adds serialization overhead just as much as this one did).

**Pull request merged**